frame/staking-rewards/src/lib.rs

```diff
@@ -780,9 +773,102 @@ pub mod pallet {
 			});
 			Ok([*position, new_position])
 		}
+
+		#[transactional]
+		fn claim(who: &Self::AccountId, position: &Self::PositionId) -> DispatchResult {
+			let keep_alive = false;
+
+			let mut stake =
+				Stakes::<T>::try_get(position).map_err(|_| Error::<T>::StakeNotFound)?;
+			ensure!(who == &stake.owner, Error::<T>::OnlyStakeOwnerCanUnstake);
+
+			let pool_id = stake.reward_pool_id;
+			let mut rewards_pool =
+				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+
+			let early_unlock = stake.lock.started_at.safe_add(&stake.lock.duration)?
```

Comment:
For claiming there is no early unlock penalty as we are not unlocking the stake so we can ignore this

Reply:
b53537bfbe37aa2fd87ad227de696d55508a12e9